src/svc.rs

```diff
@@ -12,61 +11,120 @@ pub use tower::{service_fn as mk, MakeConnection, MakeService, Service, ServiceE
 use tower_spawn_ready::SpawnReadyLayer;
 
 #[derive(Clone, Debug)]
-pub struct Builder<L>(ServiceBuilder<L>);
+pub struct Layers<L>(L);
 
-pub fn builder() -> Builder<Identity> {
-    Builder(ServiceBuilder::new())
+#[derive(Clone, Debug)]
+pub struct Stack<S>(S);
+
+pub fn layers() -> Layers<Identity> {
+    Layers(Identity::new())
+}
+
+pub fn stack<S>(inner: S) -> Stack<S> {
+    Stack(inner)
 }
 
-impl<L> Builder<L> {
-    pub fn layer<T>(self, l: T) -> Builder<Stack<T, L>> {
-        Builder(self.0.layer(l))
+impl<L> Layers<L> {
+    pub fn push<O>(self, outer: O) -> Layers<Pair<L, O>> {
+        Layers(Pair::new(self.0, outer))
     }
 
     /// Buffer requests when when the next layer is out of capacity.
-    pub fn pending(self) -> Builder<Stack<pending::Layer, L>> {
-        self.layer(pending::layer())
+    pub fn push_pending(self) -> Layers<Pair<L, pending::Layer>> {
```

Comment:
This doesn't need to be `pub` right now. It was there before, but just noticing now.